Toolkit/ArcGISToolkit/AR/ArcGISARView.swift

```diff
@@ -0,0 +1,527 @@
+//
+// Copyright 2019 Esri.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import UIKit
+import ARKit
+import ArcGIS
+
+public class ArcGISARView: UIView {
+    
+    // MARK: public properties
+    
+    /// The view used to display the `ARKit` camera image and 3D `SceneKit` content.
+    public lazy private(set) var arSCNView = ARSCNView(frame: .zero)
+    
+    /// The view used to display ArcGIS 3D content.
+    public lazy private(set) var sceneView = AGSSceneView(frame: .zero)
+    
+    /// The world tracking information used by `ARKit`.
+    public var arConfiguration: ARConfiguration = {
+        let config = ARWorldTrackingConfiguration()
+        config.worldAlignment = .gravityAndHeading
+        return config
+        }() {
+        didSet {
+            // Start tracking using the new configuration.
+            startTracking()
+        }
+    }
+
+    /// The viewpoint camera used to set the initial view of the sceneView instead of the devices GPS location via the locationManager.
+    public var originCamera: AGSCamera?
+    
+    /// The translation factor used to support a table top AR experience.
+    public var translationTransformationFactor: Double = 1.0
+    
+    // We implement ARSessionDelegate methods, but will use `sessionDelegate` to forward them to clients.
+    weak open var sessionDelegate: ARSessionDelegate?
+    
+    // We implement SCNSceneRendererDelegate methods, but will use `scnSceneRendererDelegate` to forward them to clients.
+    weak open var scnSceneRendererDelegate: SCNSceneRendererDelegate?
+
+    // MARK: private properties
+    
+    /// Whether to display the camera image or not.
+    private var renderVideoFeed = true
+    
+    /// Used to determine the device location when originCamera is not set.
+    private lazy var locationManager: CLLocationManager = {
```

Comment:
`self` really should be captured differently here:
```suggestion
    private lazy var locationManager: CLLocationManager = { [unowned self] in
```